addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -216,29 +215,39 @@ odoo.define('website_slides.quiz', function (require) {
         /**
          * @private
          * Decorate the answer inputs according to the correction
+         * and adds the answer comment if any
          */
-        _renderAnswersHighlighting: function () {
+        _renderAnswersHighlightingAndComments: function () {
             var self = this;
-            this.$('a.o_wslides_quiz_answer').each(function () {
-                var $answer = $(this);
-                var answerId = $answer.data('answerId');
-                if (_.contains(self.quiz.goodAnswers, answerId)) {
-                    $answer.removeClass('list-group-item-danger').addClass('list-group-item-success');
-                    $answer.find('i.fa').addClass('d-none');
-                    $answer.find('i.fa-check-circle').removeClass('d-none');
-                }
-                else if (_.contains(self.quiz.badAnswers, answerId)) {
-                    $answer.removeClass('list-group-item-success').addClass('list-group-item-danger');
-                    $answer.find('i.fa').addClass('d-none');
-                    $answer.find('i.fa-times-circle').removeClass('d-none');
-                    $answer.find('label input').prop('checked', false);
-                }
-                else {
-                    if (!self.slide.completed) {
-                        $answer.removeClass('list-group-item-danger list-group-item-success');
-                        $answer.find('i.fa').addClass('d-none');
-                        $answer.find('i.fa-circle').removeClass('d-none');
+            this.$('.o_wslides_js_lesson_quiz_question').each(function () {
+                var $question = $(this);
+                var questionId = $question.data('questionId');
+                var isCorrect = self.quiz.answers[questionId].is_correct;
+                $question.find('a.o_wslides_quiz_answer ').each(function () {
+                    var $answer = $(this);
+                    if ($answer.find('input[type=radio]')[0].checked) {
+                        if (isCorrect) {
+                            $answer.removeClass('list-group-item-danger').addClass('list-group-item-success');
+                            $answer.find('i.fas').addClass('d-none');
+                            $answer.find('i.fa-check-circle').removeClass('d-none');
+                        } else {
+                            $answer.removeClass('list-group-item-success').addClass('list-group-item-danger');
+                            $answer.find('i.fas').addClass('d-none');
+                            $answer.find('i.fa-times-circle').removeClass('d-none');
+                            $answer.find('label input').prop('checked', false);
+                        }
+                    } else {
+                        if (!self.slide.completed) {
```

Comment:
Why do we test this ? When completing the quiz after a failure we still see the previous failed answer, is it useful ?